services/meta/client.go

```diff
@@ -400,6 +400,37 @@ func (c *Client) User(name string) (*UserInfo, error) {
 	return nil, ErrUserNotFound
 }
 
+// UserWithPassword returns a user if the given password is correct.
+func (c *Client) UserWithPassword(username, password string) (*UserInfo, error) {
```

Comment:
At the time I was not sure if the new query should update the auth cache or not (like Authenticate does), but I don't see any reason why not now.